Dalamud/Networking/Http/HappyHttpClient.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using Dalamud.IoC;
+
+namespace Dalamud.Networking.Http;
+
+/// <summary>
+/// A service to help build and manage HttpClients with some semblance of Happy Eyeballs (RFC 8305 - IPv4 fallback)
+/// awareness.
+/// </summary>
+[PluginInterface]
```

Comment:
Don't expose this to plugins; they should manage their own `HttpClient`s.

(TBD: If we expose the `HappyEyeballsCallback`, do we want plugins to be able to use that instead?)

Reply:
Yeah, let's just expose the callback in utility namespace somewhere so people can do it if they want to, for now.